src/number/streaming.rs

```diff
@@ -1490,6 +1490,84 @@ where
   Ok((i, (sign, integer, fraction, exp)))
 }
 
+macro_rules! float_finite {
+  ($input:ident, $t:ident) => {{
+    let (i, (sign, integer, fraction, exponent)) = recognize_float_parts($input)?;
+
+    let mut float: $t = minimal_lexical::parse_float(
+      integer.as_bytes().iter(),
+      fraction.as_bytes().iter(),
+      exponent,
+    );
+    if !sign {
+      float = -float;
+    }
+
+    Ok((i, float))
+  }};
+}
+
+macro_rules! float_nonfinite {
+  ($input:ident, $t:ident) => {{
+    let (input, sign) = sign($input.clone())?;
+    let b = input.as_bytes();
+    let (mut float, count) = if b.len() >= 3 {
```

Comment:
this will not parse correctly in streaming: If we get `Na` as partial input it will return an error instead of Incomplete (once we refill the buffer, the next character could be  `N` and we get `NaN`).
The [Compare trait](https://github.com/Geal/nom/blob/master/src/traits.rs#L717) already handles those cases